mmv1/third_party/terraform/provider/provider_user_project_override_test.go

```diff
@@ -0,0 +1,414 @@
+package provider_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+// TestAccSdkProvider_user_project_override is a series of acc tests asserting how the plugin-framework provider handles credentials arguments
+// It is PF specific because the HCL used uses a PF-implemented data source
+// It is a counterpart to TestAccFwProvider_user_project_override
+func TestAccSdkProvider_user_project_override(t *testing.T) {
+	testCases := map[string]func(t *testing.T){
+		// Configuring the provider using inputs
+		"config takes precedence over environment variables":                                testAccSdkProvider_user_project_override_configPrecedenceOverEnvironmentVariables,
+		"when user_project_override is unset in the config, environment variables are used": testAccSdkProvider_user_project_override_precedenceOrderEnvironmentVariables,
+
+		// Schema-level validation
+		"when user_project_override is set in the config the value can be a boolean (true/false) or a string (true/false/1/0)": testAccSdkProvider_user_project_override_booleansInConfigOnly,
+		"when user_project_override is set via environment variables any of these values can be used: true/false/1/0":          testAccSdkProvider_user_project_override_envStringsAccepted,
+
+		// Usage
+		"user_project_override controls which project is used for quota and billing purposes":            testAccProviderUserProjectOverride,
+		"user_project_override works for resources that don't take a project id as an argument directly": testAccProviderIndirectUserProjectOverride,
```

Comment:
These are just me pulling `TestAccProviderUserProjectOverride` and `TestAccProviderIndirectUserProjectOverride` into this overall test about how user_project_override is used by and impacts the provider.

If consolidating tests this way doesn't seem like a good pattern I'm happy to receive feedback! My motivation is that the overall TestAccSdkProvider_<provider argument name> test, or at least the provider_<provider argument name>_test.go file, will group tests that are relevant to a specific provider argument and by doing this grouping we can identify missing test cases.

Reply:
The shuffle in consolidation makes sense to me, and seems like the correct decision to me.

Could you elaborate on what you mean by the latter comment? I'm not sure I follow how the being explicit = using another token for information

Reply:
Yeah sure, I was definitely not clear 😅 

In the tests I link to the provider uses both user_project_override and billing_project to signal to the provider to set the `X-Goog-User-Project` header differently, but I realised that in the acceptance tests in this PR we set user_project_override but I was unclear where the project ID used to set `X-Goog-User-Project` comes from.

When I wrote that comment I thought it was somehow derived from the access token, but that's incorrect. Instead it looks like, when billing_project is not set, the project value (from the resource's project value or the provider-default value) will be used for `X-Goog-User-Project`. For example in `TestAccProviderUserProjectOverride` we use google_kms_key_ring, and that resource [uses project if billing project is not present](https://github.com/hashicorp/terraform-provider-google-beta/blob/b1ee0e9b104c090a604d1887f2be6ddfa66f1785/google-beta/services/kms/resource_kms_key_ring.go#L116-L121).

I'll update the test case description to be more explicit about what is actually happening in the config, i.e. user_project_override does rely on other values.

Reply:
The changes in this commit are due to the above [f66b2b8](https://github.com/GoogleCloudPlatform/magic-modules/pull/11686/commits/f66b2b8aa4c1b51db5dd415e270bd36a018d1d65)